server/server.go

```diff
@@ -79,12 +80,45 @@ var dirPaths = []string{
 	"trie",
 }
 
+// set log output to the designated file
+func setLogFileWriter(logFilePath string) (*os.File, error) {
+	// if file path is empty string return error
+	if logFilePath == "" {
+		return nil, fmt.Errorf("log file path not defined")
```

Comment:
This should be defined as a package local error with `errors.New`, something like `errLogPathUndefined`.

Here, you could just return the error and check for them in the `newLogger` method explicitly

Reply:
You want the second error to be fatal, since something that was configured doesn't function like it should, you shouldn't swap it later for a CLI logger because it failed

Reply:
Refactored the logger instantiation logic fe41b1e3945aa3ff7c35ff6095786ce8d717692c